File: backend/src/domain/entities/Organization.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
} from 'typeorm';

@Entity('organizations')
export class Organization {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ type: 'varchar', nullable: true })
  name: string;

  @Column({ type: 'varchar', unique: true, nullable: true })
  slug: string;

  @Column({ type: 'varchar', name: 'logo_url', nullable: true })
  logoUrl: string;

  @Column({ type: 'json', name: 'brand_config', nullable: true })
  brandConfig: any;

  @Column({ type: 'json', name: 'features_config', nullable: true })
  featuresConfig: any;

  @CreateDateColumn({ type: 'timestamp', name: 'created_at' })
  createdAt: Date;
}

File: backend/src/main/java/com/service/backend/domain/entity/Organization.java
package com.service.backend.domain.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("organizations")
public class Organization {

    @Id
    private Long id;

    private String name;

    private String slug;

    @Column("logo_url")
    private String logoUrl;

    @Column("brand_config")
    private String brandConfig; // JSON stored as String

    @Column("features_config")
    private String featuresConfig; // JSON stored as String

    @CreatedDate
    @Column("created_at")
    private LocalDateTime createdAt;
}
